src/util/BoxSizing.hxx

```diff
@@ -0,0 +1,101 @@
+#ifndef BOX_SIZING_HXX
+#define BOX_SIZING_HXX
+
+#include "../util/Meta.hxx"
+
+/**
+ * @class       ScreenMeasurement
+ * @brief       Represents a measurement made on the screen
+ * @details     This class is used to encode linear combinations
+ *              of absolute and relative measurements
+ */
+struct ScreenMeasurement
+{
+  /** @brief    Number of pixels */
+  int px;
+  /** @brief    Percentage of an element's width */
+  int ew;
+  /** @brief    Percentage of an element's height */
+  int eh;
+  /** @brief    Percentage of a layout's width */
+  int lw;
+  /** @brief    Percentage of a layout's height */
+  int lh;
+  /** @returns  The measurement result in pixels */
+  int measure(const class ScreenMeasurementValuation &);
+};
+
+/* Literal operators */
+constexpr ScreenMeasurement operator""_lh(unsigned long long);
+constexpr ScreenMeasurement operator""_lw(unsigned long long);
+constexpr ScreenMeasurement operator""_eh(unsigned long long);
+constexpr ScreenMeasurement operator""_ew(unsigned long long);
+constexpr ScreenMeasurement operator""_px(unsigned long long);
+
+/* Arithmetic operators */
+constexpr ScreenMeasurement operator+(const ScreenMeasurement &, const ScreenMeasurement &);
+constexpr ScreenMeasurement operator-(const ScreenMeasurement &, const ScreenMeasurement &);
+
+/** 
+ * @class       ScreenMeasurementValuation
+ * @brief       Represents values of pixels to measure a ScreenMeasurement
+ */
+struct ScreenMeasurementValuation
+{
+  int elementWidth;
+  int elementHeight;
+  int layoutWidth;
+  int layoutHeight;
+};
+
+/** 
+ * @class       BoxSizing
+ * @brief       Represents the desired dimensions of a Box
+ */
+struct BoxSizing
+{
+  ScreenMeasurement width;
+  ScreenMeasurement height;
+};
+
+/**
+ * @class       PositionHint
+ * @brief       Represents a hint to the position of a (or multiple) Box
```

Comment:
Can you explain here what a "hint" is?

Reply:
What I mean by "hint" is that Layouts are free to do whatever they please. Views will provide a PositionHint to their Layout, and the Layout will try to position them as best as they can. Views should not expect that their "hint" be the ultimate position they receive unless the Layout guarantees it.